packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,253 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import classnames from 'classnames';
+
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { useRef, useState } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	FontSizePicker,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	BlockColorsStyleSelector,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import { __ } from '@wordpress/i18n';
+
+const DEFAULT_AVATAR_SIZE = 24;
```

Comment:
Is there a reason to assign this in code, rather than as the default value for the attribute?

Reply:
Good idea, it should be the default attr val.

Reply:
> Good idea, it should be the default attr val.

I looked at it briefly as part of #22359, and there was one potential "gotcha" which could explain why it was treated differently.

At this line of code, we'll only set a value if there's one from attributes, and not consider the default:

https://github.com/WordPress/gutenberg/blob/a105007ab71d33f64067878e9892a02d6a7297d8/packages/block-library/src/post-author/edit.js#L139

I can't say whether that was intentional, but I could see a case to avoid showing the default value as selected (i.e. instead, show nothing as selected until an explicit choice is made). This would behave a lot like the font size picker of Paragraph block, for example.

I still wish it were something we could do as a default attribute instead. Not only does it avoid duplication, it could probably avoid some bugs (e.g. the Post Author avatar currently won't show on the front-end if there is no size selected). 

For me, it could be:

- Just show the selected value, even if it's the default
- Or, hide the value when it's the same as the default
   - Note: This should consider if the user makes an explicit choice which coincidentally is the same as the default. Technically, these are treated as roughly equivalent, though probably not as expected (#7342).
- Or, implement some way to differentiate whether an attribute value was actually sourced from the block's original markup, or if it was provided as a default.